content/docs/learning-guide/tutorials/custom-tools/shape-example-cpp.md

```diff
@@ -0,0 +1,383 @@
+
+First, wrap these UI elements in their own sub-widget, set the layout, and add it to the main widget. 
+
+1. To start, instantiate a `QWidget` called `entityNameWidget` and a `QFormLayout` called `formLayout`.
+   
+2. Later, you will create the input field and check box and add them to this widget. 
+
+3. Finally, set the layout of `entityNameWidget` to `formLayout`, and add `entityNameWidget` to `mainLayout`. 
+
+```cpp
+QWidget* entityNameWidget = new QWidget(this);
+QFormLayout* formLayout = new QFormLayout();
+
+// Create an input field and check box
+
+entityNameWidget->setLayout(formLayout);
+mainLayout->addWidget(entityNameWidget);
+```
+
+### Add an input field
+
+Next, create an input field using the `QLineEdit` object called `m_nameInput`. Set the placeholder text in the input field using `setPlaceholderText(...)` and toggle a button to clear the text using `setClearButtonEnabled(...)`.
+
+
+```cpp
+m_nameInput = new QLineEdit(this);
+m_nameInput->setPlaceholderText(QObject::tr("Set custom Entity name here..."));
+m_nameInput->setClearButtonEnabled(true);
+```
+
+### Add a check box
+
+Then, create a check box using the `QCheckBox` object called `m_addShapeNameSuffix`. You can enable or disable the check box by default using `setDisabled(...)`.
+
+```cpp
+m_addShapeNameSuffix = new QCheckBox(this);
+m_addShapeNameSuffix->setDisabled(true);
+```
+
+### Add a signal listener with a slot handler
+
+Qt uses signals and slots to communicate between objects (refer to [Signals & Slots](https://doc.qt.io/qt-5/signalsandslots.html) in Qt Documentation). You will set up a signal listener such that when the user updates the input field, the check box automatically enables. 
+
+In this example, the signal listener uses a slot handler, which is essentially a C++ function that a signal can connect to. 
+
+1. Define a slot by using the `Q_SLOTS` macro as shown in the following line of `ShapeExampleWidget.h`:
+
+    ```cpp
+    private Q_SLOTS:
+        void OnNameInputTextChanged(const QString& text);
+    ```
+
+2. Call `QObject::connect(...)` to create a connection between the input field (`m_nameInput`) to the signal (`QLineEdit::TextChanged`) and slot (ShapeExampleWidget::OnNameINputTextChanged`). Refer to the following line of `ShapeExampleWidget.cpp`.
+
+
+```cpp
+QObject::connect(m_nameInput, &QLineEdit::textChanged, this, &ShapeExampleWidget::OnNameInputTextChanged);
+```
+
+3. Define `OnNameInputTextChanged(...)` in `ShapeExampleWidget.cpp`. This function checks if `m_nameInput` contains text, and if so, enables `m_addShapeNameSuffix`. 
+
+```cpp
+void ShapeExampleWidget::OnNameInputTextChanged(const QString& text)
+{   
+    m_addShapeNameSuffix->setEnabled(!text.isEmpty());
+}
+```
+
+
+## Buttons
+
```

Comment:
I would love to see screenshots under this headings of what the heading is referring to. E.g. Buttons